x/ccv/consumer/module.go

```diff
@@ -160,6 +160,17 @@ func (am AppModule) BeginBlock(ctx sdk.Context, req abci.RequestBeginBlock) {
 
 		// cleanup state
 		am.keeper.DeleteProviderChannel(ctx)
+		am.keeper.DeleteProviderClient(ctx)
+		am.keeper.DeletePort(ctx)
+		am.keeper.ClearCCValidators(ctx)
+		am.keeper.DeletePendingChanges(ctx)
+		am.keeper.DeleteUnbondingTime(ctx)
+		am.keeper.ClearPacketMaturityTimes(ctx)
+		am.keeper.ClearOutstandingDowntimes(ctx)
+		am.keeper.ClearHistoricalInfo(ctx)
+		am.keeper.ClearHeightValsetUpdateIDs(ctx)
+		am.keeper.ClearPendingSlashRequests(ctx)
+		am.keeper.SetParams(ctx, types.NewParams(false, 0, "", ""))
```

Comment:
Why do we need to reset the params?

Reply:
Like most of the consumer states, the params will be overwritten during `InitGenesis` if the chain is restarted as a consumer chain. So it isn't required to delete them but it shows that they won't be used anymore.